### PR TITLE
Add subset of Blob Storage E2E tests

### PR DESCRIPTION
**Pull request by epicfaace**
I've added a subset of Blob Storage E2E tests (from test_cli.py), which now work since we have both upload and download functionality working 🎉. I had to also change some logic related to hashing directories / removing directories to get the E2E tests to work.

This PR is based on https://github.com/codalab/codalab-worksheets/pull/3516, so merge that one first.

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -295,6 +295,67 @@ jobs:
...
+          # - resources
+          # - memoize
+          # - copy netcat netcurl
+          # - edit
+          # - open wopen
+    steps:
+      - name: Clear free space
+        run: |
+            sudo rm -rf /opt/ghc
+            df -h
```
Comment (adiprerepa) [inline .github/workflows/test.yml:320]: why do we run `df -h` here? do we need the output?

Comment (adiprerepa) [reply .github/workflows/test.yml:320]: ah well it exists in the rest of the test setup as well

**Review by adiprerepa — APPROVED** [review]
:confetti_ball:

File: .github/workflows/test.yml (hunk 1 above)
Comment (epicfaace) [reply .github/workflows/test.yml:320]: Yeah, don't think it's needed, but maybe we can remove in a later refactor

**Pull request merged**